docs/dev/compiler-toolchain/overview.md

```diff
@@ -0,0 +1,68 @@
+# Overview
+
+This section introduces an LLVM-based compiler toolchain for smart contract languages with Ethereum Virtual Machine
+(EVM) support. The toolchain works on top of existing compilers and requires their output, which typically includes
+intermediate representations (IRs), abstract syntax trees (ASTs), and auxiliary contract metadata and documentation.
+::: info
+At the time of writing, we support Solidity and Vyper.
+:::
+
+The toolchain consists of the following:
+
+1. [High-level source code compilers](#high-level-source-code-compilers): `solc` and `vyper`.
+2. [IR compilers, or LLVM front-end](#ir-compilers): `zksolc` and `zkvyper`.
```

Comment:
https://github.com/matter-labs/zksync-web-era-docs/blob/main/README.md uses “frontend” without a hyphen.  Either is acceptable, but we should be consistent.

Reply:
Let's sync the style. I'm not sure what is accepted in the rest of the docs.

Reply:
Commonly without hyphen - our docs are currently inconsistent by the looks.

Reply:
We should sync to using hyphen:
1. As for me, it is more readable
2. There is also `middle-end` with the same rights as `front-end` and `back-end`, as it is always spelled with a hyphen.

Reply:
disagree, we should sync to no hyphen